packages/edit-post/src/components/sidebar/post-template/create-modal.js

```diff
@@ -0,0 +1,145 @@
+/**
+ * WordPress dependencies
+ */
+import { useSelect, useDispatch } from '@wordpress/data';
+import { store as editorStore } from '@wordpress/editor';
+import { useState } from '@wordpress/element';
+import { serialize, createBlock } from '@wordpress/blocks';
+import {
+	Modal,
+	Flex,
+	FlexItem,
+	TextControl,
+	Button,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { cleanForSlug } from '@wordpress/url';
+
+/**
+ * Internal dependencies
+ */
+import { store as editPostStore } from '../../../store';
+
+const DEFAULT_TITLE = __( 'Custom Template' );
+
+export default function PostTemplateCreateModal( { onClose } ) {
+	const settings = useSelect(
+		( select ) => select( editorStore ).getEditorSettings(),
+		[]
+	);
```

Comment:
```suggestion
	const defaultBlockTemplate = useSelect(
		( select ) =>
			select( editorStore ).getEditorSettings()?.defaultBlockTemplate,
		[]
	);
```

Nitpick (non-blocking): Since we only use `defaultBlockTemplate` in the component, let's optimize the selector.